### PR TITLE
Improve build script

### PR DESCRIPTION
**Pull request by scottbommarito**
Added a `build.ps1` and `test.ps1` to replace the outdated `build.cmd`. It is modeled after the build scripts in our other repositories.

Removed `BinSkim`.

Remove the named arguments in attributes that causes issues with some parsers.

Export test results as xml

### CODE REVIEW

File: src/Search.RebuildIndex/Search.RebuildIndex.csproj [hunk 1]
```diff
@@ -139,22 +139,43 @@
       <HintPath>..\..\packages\Newtonsoft.Json.9.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
       <Private>True</Private>
     </Reference>
+    <Reference Include="NuGet.Common, Version=3.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (skofman1) [inline src/Search.RebuildIndex/Search.RebuildIndex.csproj]: why do we need those new dependencies?

**Comment by skofman1** [comment]
> @echo Off

shouldn't this file use build.ps1 and test.ps1

---
Refers to: build.cmd:1 in c23dd60. [](commit_id = c23dd601d73dc00dc8b39d718e594bdcd9504275, deletion_comment = True)

File: src/Search.RebuildIndex/Search.RebuildIndex.csproj (hunk 1 above)
Comment (scottbommarito) [inline src/Search.RebuildIndex/Search.RebuildIndex.csproj]: responded over email

**Comment by scottbommarito** [comment]
@skofman1 `build.cmd` is the old build script. For compatibility's sake (e.g. `TeamCity`) I'm leaving it in here for now.

File: src/ArchivePackages/ArchivePackages.Job.cs [hunk 2]
```diff
@@ -221,85 +221,85 @@ public class JobEventSource : EventSource
         private JobEventSource() { }
 
         [Event(
-            eventId: 1,
```
Comment (ryuyu) [inline src/ArchivePackages/ArchivePackages.Job.cs]: Split this into a separate PR so we can track the purpose of this change independently of the build script changes.

File: build/common.ps1 [hunk 3]
```diff
@@ -0,0 +1,406 @@
+### Constants ###
```
Comment (ryuyu) [inline build/common.ps1]: Is this script required to live in build folder? Any reason we can't rename it to commonBuildTasks.ps1 or something and put it next to the other build scripts? Or put all of the build scripts in build.

Comment (scottbommarito) [inline build/common.ps1]: I am following the convention that existed before. I can change the location fairly easily, but I would have to propagate it over all of the repositories however. I don't really see any benefits to moving it.

File: build.ps1 [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+    [int]$BuildNumber,
+    [switch]$SkipRestore,
+    [switch]$CleanCache,
+	[string]$SimpleVersion = '1.0.0',
+	[string]$SemanticVersion = '1.0.0-zlocal',
+	[string]$Branch,
+	[string]$CommitSHA
+)
+
+# For TeamCity - If any issue occurs, this script fail the build. - By default, TeamCity returns an exit code of 0 for all powershell scripts, even if they fail
```
Comment (xavierdecoster) [inline build.ps1]: `fail` --> `fails`

File: build.ps1 [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+$startTime = [DateTime]::UtcNow
+if (-not $BuildNumber) {
+    $BuildNumber = Get-BuildNumber
+}
+Trace-Log "Build #$BuildNumber started at $startTime"
+
+$BuildErrors = @()
+	
+Invoke-BuildStep 'Cleaning test results' { Clean-Tests } `
+	-ev +BuildErrors
```
Comment (xavierdecoster) [inline build.ps1]: Indentation seems off here (and on several other lines too).

File: build.ps1 [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+	Build-Solution $Configuration $BuildNumber -MSBuildVersion "14" $SolutionPath -SkipRestore:$SkipRestore `
+	} `
+	-args $Configuration, $BuildNumber, (Join-Path $PSScriptRoot "NuGet.Jobs.sln"), $SkipRestore `
+    -ev +BuildErrors
+	
+Invoke-BuildStep 'Prepare Validation.Callback.Vcs Package' { Prepare-Vcs-Callback } `
+	-ev +BuildErrors
+	
+Invoke-BuildStep 'Creating artifacts' {
+		New-Package (Join-Path $PSScriptRoot "src/Stats.CollectAzureCdnLogs/Stats.CollectAzureCdnLogs.csproj") -Configuration $Configuration -BuildNumber $BuildNumber -ReleaseLabel $ReleaseLabel -Version $SemanticVersion -Branch $Branch
```
Comment (xavierdecoster) [inline build.ps1]: Perhaps create an array of projects to package and loop over them, reducing the New-Package statements to a single line?

File: build/common.ps1 (hunk 3 above)
Comment (xavierdecoster) [inline build/common.ps1]: I think it is easier for first-time users of the repo to only see the actual build.cmd file in the repo root, and hide all the common scripts and tools in the `\build` folder. Not sure whether that is a standard convention, but I like it :)

File: test.ps1 [hunk 7]
```diff
@@ -0,0 +1,65 @@
+[CmdletBinding(DefaultParameterSetName='RegularBuild')]
+param (
+    [ValidateSet("debug", "release")]
+    [string]$Configuration = 'debug',
+    [int]$BuildNumber
+)
+
+# For TeamCity - If any issue occurs, this script fail the build. - By default, TeamCity returns an exit code of 0 for all powershell scripts, even if they fail
```
Comment (xavierdecoster) [inline test.ps1]: `fail` > `fails`
